frontend/simple/model/contracts/group.js

```diff
@@ -109,63 +109,81 @@ export default DefineContract({
   },
   'GroupProposal': {
     constants: {
-      TypeInvitation: 'invitationProposal',
-      TypeRemoval: 'removalProposal',
-      TypeChange: 'changeProposal'
+      // what the proposal does
+      // if updating group field, this should be the field name to update
+      DoInvitation: 'invitation',
+      DoRemoval: 'removal',
+      DoIncome: 'incomeProvided',
+      DoChangeThreshold: 'changeThreshold',
+      DoApprovalThreshold: 'memberApprovalThreshold',
+      DoRemovalThreshold: 'memberRemovalThreshold',
+      // how the proposal vote gets decided
+      // TODO better place for these? + other types?
+      RuleTypeThreshold: 'ruleTypeThreshold'
```

Comment:
Indeed, these should be placed in a different file, e.g. `model/votingRules.js`:

```js
export const VotingRuleThreshold = 'votingRuleThreshold'

export default {
  [VotingRuleThreshold]: function (parameters: Object, votes: Array<*>): number {
     // impl...
  }
}